pkg/workerapi/workerserver/server.go

```diff
@@ -48,24 +49,24 @@ func (s *server) Serve(bindAddress string) error {
 		cmux.HTTP2MatchHeaderFieldSendSettings("content-type", "application/grpc"))
 	httpListener := mux.Match(cmux.Any())
 
-	logIfErrored := func(protocol string, f func() error) {
-		if err := f(); err != nil {
+	logIfErrored := func(protocol string, err error) {
+		if err != nil && !errors.Is(err, net.ErrClosed) && !errors.Is(err, cmux.ErrListenerClosed) {
 			log.Error().WithError(err).Messagef("Error during serving %s.", protocol)
 		}
 	}
 
-	go logIfErrored("gRPC", func() error { return serveGRPC(s.grpc, grpcListener) })
-	go logIfErrored("HTTP", func() error { return serveHTTP(s.rest, httpListener) })
-
-	return mux.Serve()
+	go logIfErrored("mux", mux.Serve())
+	go logIfErrored("HTTP", serveHTTP(s, s.rest, httpListener))
+	return serveGRPC(s.grpc, grpcListener)
```

Comment:
Suggest reverting this. It needs to be passed in as a func here because the arguments to a `go` call are evaluated syncronously and not inside the new goroutine, so at line 58 it will now hault while serving the mux on the main thread, and then when it exits start `logIfErrored` inside a new goroutine, and then continue to line 59 and call `serveHTTP`...

Reply:
Reverted.